NetworkConfigurations/TrafficControlAndPolicingSubCategories/TrafficShaping.md

```diff
@@ -0,0 +1,506 @@
+
+In DENT, congestion avoidance is implemented using the RED queuing discipline, configured on the ETS qdisc bands. For more information, see [tc-red(8)](https://man7.org/linux/man-pages/man8/tc-red.8.html).
+
+On the hardware, the algorithm is implemented using a WRED (Weighted Random Early Detection) mechanism.
+
+The following parameters of the RED qdisc are offloaded:
+
+- **min**: The minimum queue size.
+- **max**: The maximum limit.
+- **probability**: The probability of dropping a packet when the average queue size reaches the maximum limit. A probability of 1.0 means 100%.
+
+**Note**: Other parameters are not offloaded, but they may be required to be provided by the user.
+
+To see RED statistics on each band (ETS bands and its child RED qdiscs), use the `qdisc show` command on the port. This command displays the current qdisc configuration, including queue statistics:
+
+```
+tc -s qdisc show dev <interface>
+```
+
+**Example Configuration:**
+
+The following example attaches a RED qdisc under band 0 of an ETS parent whose handle is 10. Between the queue depths of 500KiB and 1.5MiB, the probability of dropping packets gradually rises from 0 to 10%.
+
+```
+tc qdisc add dev swp26 root handle 10: htb default 1
+tc class add dev swp26 parent 10: classid 10:1 htb rate 1Gbit
+
+tc qdisc add dev swp26 parent 10:1 handle 101: red limit 2M avpkt 1000 probability 0.1 min 500K max 1.5M
+```
+
+```
+RED: set bandwidth to 10Mbit
+```
+
+The output confirms that the bandwidth for RED qdisc has been successfully set to 10Mbit.
+
+To view the current qdisc configuration and queue statistics:
+
+```
+tc -s qdisc show dev swp26
+```
+
+```
+qdisc htb 10: root refcnt 2 r2q 10 default 0x1 direct_packets_stat 0 direct_qlen 1000
+ Sent 0 bytes 0 pkt (dropped 0, overlimits 0 requeues 0)
+ backlog 0b 0p requeues 0
+qdisc red 101: parent 10:1 limit 2Mb min 500Kb max 1536Kb
+ Sent 0 bytes 0 pkt (dropped 0, overlimits 0 requeues 0)
+ backlog 0b 0p requeues 0
+  marked 0 early 0 pdrop 0 other 0
+```
+
+The output displays the RED qdisc configuration under the HTB parent, including the limit, min and max queue sizes, and drop probabilities.
+
+Verfication using `iperf` utility:
+
+```
+iperf -c 127.0.0.1 -t 10 -b 10M
+```
+
+Please make sure to run your iperf server using `iperf -s` command.
+
+Output-
+
+```
+------------------------------------------------------------
+Client connecting to 127.0.0.1, TCP port 5001
+TCP window size: 2.50 MByte (default)
+------------------------------------------------------------
+[  3] local 127.0.0.1 port 51748 connected with 127.0.0.1 port 5001
+[ ID] Interval       Transfer     Bandwidth
+[  3]  0.0-10.0 sec  11.9 MBytes  10.0 Mbits/sec
+```
+
+The test was executed successfully, showing a transfer of approximately 11.9 Megabytes of data at an average bandwidth of 10.0 Mbps, which aligns with the configured bandwidth of 10 Mbps.
+
```

Comment:
Similar comments as the previous three. I do not know if these examples are valid.